src/shell.rs

```diff
@@ -22,6 +28,43 @@ impl<'a> Shell<'a> {
     pub fn path_for(&self, filename: &str) -> PathBuf {
         self.root_path().join(filename)
     }
+
+    pub fn create_links<P: AsRef<Path>>(&self, base_directory: P) -> FileOperations {
+        let mut file_ops = FileOperations::rooted_at(base_directory);
+        let entries = fs::read_dir(self.root_path()).unwrap();
+        let shell_path_length = self.root_path().to_str().unwrap().len();
+        let file_root_length = file_ops.root.to_str().unwrap().len();
+        for entry in entries {
+            let entry = entry.unwrap().path();
+            let entry = entry.to_str().unwrap();
+            let rel_shell = &entry[shell_path_length+1..entry.len()];
+            let rel_file_root = &entry[file_root_length+1..entry.len()];
```

Comment:
I'm not 100% sure, but could what you're doing here with the string slicing possibly be accomplished with `strip_prefix` on `PathBuf`? If possible, I'd like to take advantage of the `Path` and `PathBuf` structs and their operations to do this kind of manipulation rather than treating them as raw strings.